src/containers/Review/ProgressBar.tsx

```diff
@@ -1,22 +1,57 @@
-import React, { useState, useEffect } from 'react'
-import { Container, Message } from 'semantic-ui-react'
-import { useUserState } from '../../contexts/UserState'
+import React from 'react'
+import { Icon, SemanticICONS } from 'semantic-ui-react'
 import { FullStructure } from '../../utils/types'
-import { useLanguageProvider } from '../../contexts/Localisation'
+// @ts-ignore -- no types provided
+import { Stepper, Step, StepTitle } from 'react-progress-stepper'
+import { ApplicationOutcome } from '../../utils/generated/graphql'
 
 const ReviewProgress: React.FC<{
   structure: FullStructure
 }> = ({ structure: fullStructure }) => {
-  const { strings } = useLanguageProvider()
   const {
-    userState: { currentUser },
-  } = useUserState()
+    stages,
+    info: {
+      outcome,
+      current: { stage: currentStage },
+    },
+  } = fullStructure
+
+  const [currentStageClass, currentStageIcon]: [string, SemanticICONS] =
+    outcome === ApplicationOutcome.Rejected ? ['rejected', 'remove'] : ['', 'circle']
+
+  const isApproved = outcome === ApplicationOutcome.Approved
+
+  const steps = [
+    <Step key="first">
+      <StepTitle>Submitted</StepTitle>
+    </Step>,
+    ...stages.map((stage) =>
+      stage.number === currentStage.number && !isApproved ? (
+        <Step
+          key={stage.name}
+          customContent={() => (
+            <div className={`step-custom-content ${currentStageClass}`}>
+              <Icon name={currentStageIcon} />
+            </div>
+          )}
+        >
+          <StepTitle>{stage.name}</StepTitle>
+        </Step>
+      ) : (
+        <Step key={stage.name}>
+          <StepTitle>{stage.name}</StepTitle>
+        </Step>
+      )
+    ),
+  ]
 
   return (
     <div id="review-progress">
-      <Message size="huge">
-        <Message.Header>Placeholder for PROGRESS BAR</Message.Header>
-      </Message>
+      <div id="stepper-wrapper">
+        <Stepper id="stepper" step={currentStage.number + Number(isApproved)} numbered={false}>
```

Comment:
Still trying to understand what this means: `step={currentStage.number + Number(isApproved)}`

Reply:
`step` defines the "current" step for the progress bar, so mostly the step number is the stage number. However, in the case where the application is completed, we want the "final" step to show as "done", so we make the progress bar "step" a higher number than there is available, so it shows ALL steps as done. So the `Number(isApproved` just adds 1 to the step number if the application is completed in order to achieve this.